File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for (const auto& command : commands)
    {
        int i = command[0];
        int j = command[1];
        int k = command[2];
        
        vector<int> tmpArray(array.begin() + i - 1, array.begin() + j);
        sort(tmpArray.begin(), tmpArray.end());
        answer.push_back(tmpArray[k - 1]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int[] solution(int[] array, int[,] commands) {
        List<int> answer = new List<int>();
        
        for (int i = 0; i < commands.GetLength(0); ++i)
        {
            int start = commands[i, 0];
            int end = commands[i, 1];
            int k = commands[i, 2];
            
            var temp = array.Skip(start - 1).Take(end - start + 1).OrderBy(x => x).ToList();
            
            answer.Add(temp[k - 1]);
        }
        
        return answer.ToArray();
    }
}